src/main/java/org/terasology/climateConditions/BodyTemperatureSystem.java

```diff
@@ -0,0 +1,134 @@
+import org.terasology.logic.delay.PeriodicActionTriggeredEvent;
+import org.terasology.logic.location.LocationComponent;
+import org.terasology.registry.In;
+import org.terasology.world.WorldComponent;
+
+
+@RegisterSystem(value = RegisterMode.AUTHORITY)
+public class BodyTemperatureSystem extends BaseComponentSystem {
+    public static final String BODY_TEMPERATURE_UPDATE_ACTION_ID = "climateConditions:bodyTemperatureUpdate";
+
+    private static final Logger logger = LoggerFactory.getLogger(BodyTemperatureSystem.class);
+
+    @In
+    private EntityManager entityManager;
+    @In
+    ClimateConditionsSystem climateConditionsSystem;
+    @In
+    DelayManager delayManager;
+
+    private static final int CHECK_INTERVAL = 1000;
+
+    public void postBegin() {
+        boolean processedOnce = false;
+        for (EntityRef entity : entityManager.getEntitiesWith(WorldComponent.class)) {
+            if (!processedOnce) {
+                delayManager.addPeriodicAction(entity, BODY_TEMPERATURE_UPDATE_ACTION_ID, 0, CHECK_INTERVAL);
+                processedOnce = true;
+            } else {
+                logger.warn("More than one entity with WorldComponent found");
+            }
+        }
+        // The world is the entity here which shall have a Body Temperature Update Periodic Action.
+    }
+
+    @ReceiveEvent
+    public void onTemperatureUpdate(PeriodicActionTriggeredEvent event, EntityRef world) {
+        if (event.getActionId().equals(BODY_TEMPERATURE_UPDATE_ACTION_ID)) {
+            for (EntityRef entity : entityManager.getEntitiesWith(AliveCharacterComponent.class,
+                    BodyTemperatureComponent.class, LocationComponent.class)) {
+                LocationComponent location = entity.getComponent(LocationComponent.class);
+                BodyTemperatureComponent bodyTemperature = entity.getComponent(BodyTemperatureComponent.class);
+                float envTemperature = climateConditionsSystem.getTemperature(location.getLocalPosition());
+                float envHumidity = climateConditionsSystem.getHumidity(location.getLocalPosition());
+                float deltaTemp =
+                        ((((envTemperature - (envHumidity / 10)) - bodyTemperature.current) / 100000) * CHECK_INTERVAL);
+
+                //Send event for other systems to modify change in body temperature.
+                if(deltaTemp >= 0) {
+                    AffectBodyTemperatureEvent affectBodyTemperatureEvent = new AffectBodyTemperatureEvent(deltaTemp);
+                    entity.send(affectBodyTemperatureEvent);
+                    deltaTemp = affectBodyTemperatureEvent.getResultValue();
+                } else {
+                    AffectBodyTemperatureEvent affectBodyTemperatureEvent = new AffectBodyTemperatureEvent(-1 * deltaTemp);
+                    entity.send(affectBodyTemperatureEvent);
+                    deltaTemp = -1 * affectBodyTemperatureEvent.getResultValue();
+                }
```

Comment:
Can the `affectBodyTemperatureEvent` only work with positive input? Or why do we do this here?

Reply:
Yeah AbstractValueModifiableEvent doesn't work with negative values... but right now I'm making some changes so that at least the AffectBodyTemperature would work without doing this..